src/main/java/com/blamejared/contenttweaker/api/blocks/IIsCoTBlock.java

```diff
@@ -1,6 +1,101 @@
 package com.blamejared.contenttweaker.api.blocks;
 
-import com.blamejared.contenttweaker.api.*;
-import net.minecraftforge.common.extensions.*;
+import com.blamejared.contenttweaker.actions.ActionSetFunction;
+import com.blamejared.contenttweaker.api.IHasCoTItem;
+import com.blamejared.contenttweaker.api.IHasResourceLocation;
+import com.blamejared.contenttweaker.api.IHasResourcesToWrite;
+import com.blamejared.contenttweaker.api.functions.*;
+import com.blamejared.contenttweaker.api.items.IIsCotItem;
+import com.blamejared.crafttweaker.api.annotations.ZenRegister;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import net.minecraftforge.common.extensions.IForgeBlock;
+import org.openzen.zencode.java.ZenCodeType;
 
-public interface IIsCoTBlock extends IHasCoTItem, IHasResourcesToWrite, IHasResourceLocation, IForgeBlock {}
+/**
+ * A registered CoT Block. Used for advanced functionality. like onRandomTick, onReplaced etc.
+ * <p>
+ * These functions should be run in CraftTweaker scripts, instead of ContentTweaker ones. And they are reloadable.
+ * You can get it via cotItem BEP.
+ */
+@ZenRegister
+@Document("mods/contenttweaker/API/block/IIsCotBlock")
+@ZenCodeType.Name("mods.contenttweaker.block.IIsCotBlock")
+public interface IIsCoTBlock extends IHasCoTItem, IHasResourcesToWrite, IHasResourceLocation, IForgeBlock {
+    /**
+     * Sets what will happen when the block is added.
+     *
+     * @return the IIsCoTBlock, used for method chaining
+     */
+    @ZenCodeType.Method
+    default IIsCoTBlock setOnAdded(IBlockAdded func) {
+        ActionSetFunction.applyNewAction(func, IBlockAdded.class, this);
+        return this;
+    }
+
+    /**
+     * Sets what will happen when neighbor of the block is changed
+     *
+     * @return the IIsCoTBlock, used for method chaining
+     */
+    @ZenCodeType.Method
+    default IIsCoTBlock setOnNeighborChanged(IBlockNeighborChanged func) {
+        ActionSetFunction.applyNewAction(func, IBlockNeighborChanged.class, this);
+        return this;
+    }
+
+    /**
+     * Sets what will happen when the block receive a random tick. Throws an exception if the block is not ticks randomly.
+     *
+     * @return the IIsCoTBlock, used for method chaining
+     */
+    @ZenCodeType.Method
+    default IIsCoTBlock setOnRandomTick(IBlockRandomTick func) {
+        if (!this.getBlock().ticksRandomly(this.getBlock().getDefaultState())) {
+            throw new UnsupportedOperationException("You should set the block ticks randomly first! Add `withTickRandomly` to linked block builder.");
```

Comment:
Why not just set it automatically?

Reply:
Setting it tick randomly is BlockBuilder stuff, requiring CoT loader. Not there.

Reply:
I think that method just sets Data in the block which could be override in the block to avoid the confusing requirement IMO.

Reply:
There is not a tick randomly setter. The field only can be set on constuctor.

Reply:
The method you are calling can be overridden on block (Which are calling, but should only be calling form the BlockState anyway.

Reply:
override? Are you sure? It needs a new class. These function are executed on CrT side, when all register things are done.